### PR TITLE
Add columns - FunctionInvocationId, ScriptHostInstanceId and HostId

### LINKED ISSUE #2312: [V1]Include FunctionInvocationId and Host InstanceId directly in system logs
Querying/correlation would be much nicer if the function id and host instance id was a queryable column.

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #2312 
Updated all the Tracewriter.Info and Tracewriter.Error calls to include:  FunctionInvocationId, ScriptHostInstanceId and HostId.

### CODE REVIEW

File: src/WebJobs.Script/ScriptConstants.cs [hunk 1]
```diff
@@ -24,12 +24,19 @@ public static class ScriptConstants
...
+        public const string TracePropertyFunctionInvocationIdKey = "MS_FunctionInvocationId";
+        public const string TracePropertyHostIdKey = "MS_HostId";
+        public const string TracePropertyScriptHostInstanceIdKey = "MS_ScriptHostInstanceId";
 
         public const string TraceSourceSecretManagement = "SecretManagement";
         public const string TraceSourceHostAdmin = "HostAdmin";
         public const string TraceSourceFileWatcher = "FileWatcher";
         public const string TraceSourceSwagger = "SwaggerDocumentation";
         public const string TraceSourceHttpHandler = "HttpRequestTraceHandler";
+        public const string TraceSourceScriptHost = "Script.Host";
```
Comment (pragnagopa) [inline src/WebJobs.Script/ScriptConstants.cs]: @fabiocav - Any ideas on what should be the "source" column value in our logs? Do we want to see similar to what [webjobs sdk](https://github.com/Azure/azure-webjobs-sdk/blob/663a508e8a851629c26a51e7de3af36629dfd120/src/Microsoft.Azure.WebJobs.Host/TraceSource.cs) has like "Script.Host" or should it be the class name like [this](https://github.com/Azure/azure-functions-host/blob/v1.x/src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs#L196)

**Comment by mathewc** [comment]
What is the context on this - is there a work item/issue?

File: src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs [hunk 2]
```diff
@@ -43,7 +45,8 @@ internal class FunctionInstanceLogger : IAsyncCollector<FunctionInstanceLogEntry
                 var tableProvider = LogFactory.NewLogTableProvider(client);
 
                 string containerName = Environment.MachineName;
-                this._writer = LogFactory.NewWriter(hostName, containerName, tableProvider, (e) => OnException(e, trace));
+                _hostId = hostInstanceId;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs]: This naming is confusing. There are HostIds and HostInstanceIds. Which is this?

**Comment by pragnagopa** [comment]
Updated the PR description to include issue<https://github.com/Azure/azure-functions-host/issues/2312>.

**Comment by pragnagopa** [comment]
Fixed naming and added both columns.

File: src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs [hunk 3]
```diff
@@ -43,7 +48,9 @@ internal class FunctionInstanceLogger : IAsyncCollector<FunctionInstanceLogEntry
                 var tableProvider = LogFactory.NewLogTableProvider(client);
 
                 string containerName = Environment.MachineName;
-                this._writer = LogFactory.NewWriter(hostName, containerName, tableProvider, (e) => OnException(e, trace));
+                _hostId = hostId;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs]: Wondering why the HostID needs to be logged. We're already logging the AppName column, and in Functions we default the HostID to the AppName. How would this be used - do we actually need it?

**Comment by pragnagopa** [comment]
Closing this PR. Synced with @mathewc. Will add new HostInstanceId for identifying ScriptHost instance. Existing InstanceId in ScriptHost.cs reflects the website_instanceid.

**Pull request closed without merging**